test/upgrade/prober/wathola/sender/services.go

```diff
@@ -104,15 +116,52 @@ func NewCloudEvent(data interface{}, typ string) cloudevents.Event {
 	return e
 }
 
+// ResetEventSenders will reset configured event senders to defaults.
+func ResetEventSenders() {
+	eventSenders = make([]EventSender, 0, 1)
+}
+
+// RegisterEventSender will register a EventSender to be used.
+func RegisterEventSender(es EventSender) {
+	eventSenders = append(eventSenders, es)
+}
+
 // SendEvent will send cloud event to given url
-func SendEvent(e cloudevents.Event, url string) error {
+func SendEvent(ce cloudevents.Event, endpoint interface{}) error {
+	senders := make([]EventSender, 0, len(eventSenders)+1)
+	senders = append(senders, eventSenders...)
+	if len(senders) == 0 {
+		senders = append(senders, httpSender{})
+	}
+	for _, eventSender := range senders {
+		if eventSender.Supports(endpoint) {
+			return eventSender.SendEvent(ce, endpoint)
+		}
+	}
+	return fmt.Errorf("%w: endpoint is %#v", ErrEndpointTypeNotSupported, endpoint)
```

Comment:
I think the convention is to place the error at the end:
```
return fmt.Errorf("endpoint is %#v: %w", endpoint, ErrEndpointTypeNotSupported)
```

Reply:
Yup, but that's the case if we are wrapping error from other package, the cause, like that:

```golang
if err := other.Something(); err != nil {
  return fmt.Errorf("%w: %v", ErrEndpointTypeNotSupported, err)
}
```
Then resulting error can be checked with:

```golang
errors.Is(err, ErrEndpointTypeNotSupported) == true
```
Error message is: `endpoint type is not supported: cause from other.Something() function`

In this case I'm not wrapping another package errors, but creating my own error of type `ErrEndpointTypeNotSupported` and I want the message to be in form: `endpoint type is not supported: endpoint is nil`. I think the message in other order makes less sense.

Reply:
OK. I got confused by `%w`. In your case it should probably be `%v` or `%s` because you're interested only in the string itself?

Reply:
I'm interested in both correct type `ErrEndpointTypeNotSupported` (thus `%w` usage) and correct message: `"endpoint type is not supported: endpoint is nil"`. 

Current code does exaclly that. It wraps `ErrEndpointTypeNotSupported` error with new one adding detail about why this error is returned: `"endpoint is %#v"`.